packages/apps/shopify-app-remix/src/server/authenticate/admin/helpers/__tests__/redirect.test.ts

```diff
@@ -172,6 +173,53 @@ describe('Redirect helper', () => {
       );
     });
 
+    it('parses shopify admin routes', async () => {
+      // GIVEN
+      const shopify = shopifyApp(testConfig());
+      await setUpValidSession(shopify.sessionStorage);
+
+      const {request} = documentLoadRequest(true);
+      const {redirect} = await shopify.authenticate.admin(request);
+
+      // WHEN
+      const response = await getThrownResponse(
+        async () => redirect('shopify://admin/products/', {target: '_top'}),
+        request,
+      );
+
+      // THEN
+      await assertAppBridgeScript(
+        response,
+        `https://admin.shopify.com/store/${TEST_SHOP_NAME}/products/`,
+        '_top',
+      );
+    });
+
+    it('parses shopify admin routes and removes embedded params', async () => {
```

Comment:
Ideally, I think we should also have a test that leaves non-restricted params alone to ensure we don't have any regressions.